core/src/main/java/org/elasticsearch/action/search/SearchRequestBuilder.java

```diff
@@ -293,6 +293,18 @@ public SearchRequestBuilder addDocValueField(String name) {
     }
 
     /**
+     * Adds a docvalue based field to load and return. The field does not have to be stored,
+     * but its recommended to use non analyzed fields.
+     *
+     * @param name The field to get from the docvalue
+     * @param format How to format the field, {@code null} to use defaults.
+     */
+    public SearchRequestBuilder addDocValueField(String name, String format) {
```

Comment:
I know Kibana uses this feature to retrieve dates in millis since Epoch, so with this option they will be able to pass `format: epoch_millis` when we format docvalue fields. I also plan to use it for the transition period: users will be able in 5.x to enable the 6.x behaviour by using the special format `use_field_defaults`.